File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
	people.sort()
	people=deque(people)
	answer=0
	while len(people)>1:
		if people[0]+people[-1]>limit:
			people.pop()
			answer+=1
		else:
			answer+=1
			people.popleft()
			people.pop()
	answer+=len(people)
	return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] people, int limit) {
        ArrayDeque<Integer> sortedDeque = Arrays.stream(people)  // 배열을 스트림으로 변환
                                                .sorted()        // 정렬
                                                .boxed()         // int를 Integer로 박싱
                                                .collect(Collectors.toCollection(ArrayDeque::new));  
        // 결과를 ArrayDeque로 수집
        int answer=0;
        while(!sortedDeque.isEmpty()){
            if(sortedDeque.peek()+sortedDeque.peekLast()>limit){
                sortedDeque.pollLast();
                answer++;
            }else{
                answer++;
                sortedDeque.pollLast();
                sortedDeque.poll();
            }
        }
        answer+=sortedDeque.size();
        
        

        return answer;
    }
}